### PR TITLE
Replace focus() with focusWithKeyboard()

### PR DESCRIPTION
**Pull request by dlockhart**
This builds on [the previous change](https://github.com/BrightspaceUI/core/pull/3385), this time switching visual-diff uses of `focus()` to `focusWithKeyboard()`.

I've left 8 occurrences of `focus()` from the date/time input components, simply due to them having some complex relationships between the focus code and events. [Here's an example](https://github.com/BrightspaceUI/core/blob/9f91e8b53818282a04e241f4886de6a1e1a8a8cf/components/inputs/test/input-date-range.visual-diff.js#L185), where a nested shadow root query is done to wire up an event listener that resolves a Promise and the `focus()` call initiates that. If anyone has ideas about how to solve something like this using `focusWithKeyboard`, I'm all 👂s!

### CODE REVIEW

File: components/list/test/screenshots/ci/golden/list/d2l-list-expand-collapse-button-focus.png
Comment (dlockhart) [inline components/list/test/screenshots/ci/golden/list/d2l-list-expand-collapse-button-focus.png:1]: These two changes are expected -- we're now getting proper keyboard focus styles compared to before. 🎉

File: components/tabs/test/screenshots/ci/golden/tabs/d2l-tabs-basic-selected-tab-focus.png
Comment (dlockhart) [inline components/tabs/test/screenshots/ci/golden/tabs/d2l-tabs-basic-selected-tab-focus.png:1]: Also expecting these two changes now that focus is happening with the keyboard properly.

**Comment by eKoopmans** [comment]
Still looking at this PR 👀, got split focus (pun intended?) right now with axe-con.

> I've left 8 occurrences of focus() from the date/time input components, simply due to them having some complex relationships between the focus code and events. [Here's an example](https://github.com/BrightspaceUI/core/blob/9f91e8b53818282a04e241f4886de6a1e1a8a8cf/components/inputs/test/input-date-range.visual-diff.js#L185), where a nested shadow root query is done to wire up an event listener that resolves a Promise and the focus() call initiates that. If anyone has ideas about how to solve something like this using focusWithKeyboard, I'm all 👂s!

For that example, here's my suggestion:

```js
		async function focusOnInput(page, selector, inputSelector) {
			await page.$eval(selector, (elem, inputSelector) => {
				elem.blur();
				const input = elem.shadowRoot.querySelector(inputSelector);
				return new Promise((resolve) => {
					elem.addEventListener('d2l-tooltip-show', resolve, { once: true });
				});
			}, inputSelector);
			await focusWithKeyboard(page, [selector, inputSelector]);
		}
```

**Comment by dlockhart** [comment]
No worries! Unfortunately that won't work since focusing is the thing that fires the `d2l-tooltip-show` event, so the Promise from the first async call won't ever resolve.

I think I can accomplish this by re-ordering them and doing the focus in a different thread:
```javascript
setTimeout(() => focusWithKeyboard(page, [selector, inputSelector]));
await page.$eval(selector, (elem, inputSelector) => {
	elem.blur();
	const input = elem.shadowRoot.querySelector(inputSelector);
	return new Promise((resolve) => {
		elem.addEventListener('d2l-tooltip-show', resolve, { once: true });
	});
}, inputSelector);
```

That approach worked elsewhere, but I'm not sure if it's introducing a timing-related flake.

**Comment by eKoopmans** [comment]
Ah interesting, I missed that. How about:

```js
function promiseWithResolve() {
	let promiseResolve;
	const promise = new Promise(resolve => promiseResolve = resolve);
	return Object.assign(promise, { resolve: promiseResolve });
}

		async function focusOnInput(page, selector, inputSelector) {
			const tooltipShown = promiseWithResolve();
			await page.$eval(selector, (elem, inputSelector) => {
				elem.blur();
				const input = elem.shadowRoot.querySelector(inputSelector);
				elem.addEventListener('d2l-tooltip-show', tooltipShown.resolve, { once: true });
			}, inputSelector);
			await focusWithKeyboard(page, [selector, inputSelector]);
			await tooltipShown;
		}
```

**Comment by dlockhart** [comment]
Won't work either since `tooltipShown` would need to be passed into `$eval` as a parameter, and only JSON serializable object types or JSHandles are supported.

File: components/tabs/test/tabs.visual-diff.js [hunk 1]
```diff
@@ -7,12 +7,10 @@ describe('d2l-tabs', () => {
...
-	const focusTabs = (selector) => {
-		return page.$eval(selector, (elem) => {
-			return new Promise((resolve) => {
-				elem.focus();
-				requestAnimationFrame(resolve);
-			});
+	const focusTabs = async(selector) => {
+		await focusWithKeyboard(page, selector);
+		await page.evaluate(() => {
+			new Promise(resolve => requestAnimationFrame(resolve));
```
Comment (eKoopmans) [inline components/tabs/test/tabs.visual-diff.js:13]: So, some cases of needing `requestAnimationFrame`...

**Review by eKoopmans — APPROVED** [review]

File: components/tabs/test/tabs.visual-diff.js (hunk 1 above)
Comment (dlockhart) [reply components/tabs/test/tabs.visual-diff.js:13]: Yeah, not a new case but another case still the same.

Comment (dbatiste) [reply components/tabs/test/tabs.visual-diff.js:13]: As you might know, tabs does a lot of stuff when moving focus around, and those things are `async`, which is probably why rAF is needed.

**Review by dbatiste — APPROVED** [review]

**Pull request merged**